rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/NodesRecoveryManager.java

```diff
@@ -25,48 +25,72 @@
+import org.objectweb.proactive.core.node.Node;
 import org.ow2.proactive.resourcemanager.authentication.Client;
+import org.ow2.proactive.resourcemanager.common.NodeState;
+import org.ow2.proactive.resourcemanager.core.properties.PAResourceManagerProperties;
+import org.ow2.proactive.resourcemanager.db.NodeSourceData;
+import org.ow2.proactive.resourcemanager.db.RMNodeData;
+import org.ow2.proactive.resourcemanager.nodesource.NodeSource;
+import org.ow2.proactive.resourcemanager.nodesource.NodeSourceDescriptor;
+import org.ow2.proactive.resourcemanager.nodesource.NodeSourceStatus;
+import org.ow2.proactive.resourcemanager.nodesource.infrastructure.InfrastructureManager;
+import org.ow2.proactive.resourcemanager.nodesource.infrastructure.InfrastructureManagerFactory;
 import org.ow2.proactive.resourcemanager.rmnode.RMNode;
 
 import com.google.common.base.Function;
 
 
 /**
- * This classe handles the recovery of node sources and of nodes when the RM is restarted.
+ * This class handles the recovery of node sources and of nodes when the RM is restarted.
  *
  * @author ActiveEon Team
  * @since 26/06/17
  */
 public class NodesRecoveryManager {
 
+    public static final String START_TO_RECOVER_NODES = "Start to recover nodes";
+
+    public static final String END_OF_NODES_RECOVERY = "Total number of nodes recovered: ";
+
     private static final Logger logger = Logger.getLogger(NodesRecoveryManager.class);
 
-    private RMCore resourceManager;
+    private RMCore rmCore;
 
     private NodesLockRestorationManager nodesLockRestorationManager;
 
-    public NodesRecoveryManager(RMCore rmCore) {
-        resourceManager = rmCore;
+    protected NodesRecoveryManager(RMCore rmCore) {
+        this.rmCore = rmCore;
     }
 
-    public void initialize() {
-        initNodesRestorationManager();
+    protected void initialize() {
+        this.initNodesRestorationManager();
     }
 
-    void initNodesRestorationManager() {
-        nodesLockRestorationManager = getNodesLockRestorationManagerBuilder().apply(resourceManager);
+    protected void initNodesRestorationManager() {
+        this.nodesLockRestorationManager = this.getNodesLockRestorationManagerBuilder().apply(this.rmCore);
 
-        if (RM_NODES_LOCK_RESTORATION.getValueAsBoolean()) {
-            nodesLockRestorationManager.initialize();
+        if (PAResourceManagerProperties.RM_NODES_LOCK_RESTORATION.getValueAsBoolean()) {
+            this.nodesLockRestorationManager.initialize();
         } else {
             logger.info("Nodes lock restoration is disabled");
         }
     }
 
-    Function<RMCore, NodesLockRestorationManager> getNodesLockRestorationManagerBuilder() {
+    protected void restoreLock(RMNode rmNode, Client provider) {
+        this.nodesLockRestorationManager.handle(rmNode, provider);
+    }
+
+    protected Function<RMCore, NodesLockRestorationManager> getNodesLockRestorationManagerBuilder() {
         return new Function<RMCore, NodesLockRestorationManager>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)